### PR TITLE
Add shaker temperature gauging

### PR DESCRIPTION
**Pull request by TDHooligan**
[QOL][SPRITES][OBJECTS]
## About the PR 

Shakers show temperature of contained drinks, with sprites indicating when they're in unsafe drinking ranges.
Hot: >97C (Burn damage)
Warm: >30C (icky beer)
Cold: <7C (cool beer)
Freezing: <-23C (Burn damage)

image
image
Sprites by Guv. Originally in #18351 

Also fixes to stop infinite recursion of temperature changes in menders.

## Why's this needed? 
Better gauging of temperature for barkeeping. More reasons for cocktail shaker existing.

## Changelog 

```changelog
(u)TDHooligan, Guv
(+)The bartender's shaker now shows if your cocktail is a safe temperature to drink
```

### CODE REVIEW

**Review by TobleroneSwordfish — APPROVED** [review]
Lgtm, it might be nice to extend something like this to all reagent containers even if it's just a line of description like "it feels hot to the touch"

File: code/modules/chemistry/tools/patches.dm [hunk 1]
```diff
@@ -477,7 +477,7 @@ TYPEINFO(/obj/item/reagent_containers/mender)
 
 	on_reagent_change(add)
 		..()
-		if (src.reagents)
+		if (src.reagents && src.reagents.total_temperature > 330)
```
Comment (Tarmunora) [inline code/modules/chemistry/tools/patches.dm:480]: ?

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/modules/chemistry/tools/patches.dm (hunk 1 above)
Comment (TDHooligan) [reply code/modules/chemistry/tools/patches.dm:480]: infinite loop triggered by adding reagents_changed callbacks to temperature_reagents. 

makes more sense that heating reagents also counts as changing them, better place to hook it.

Comment (Tarmunora) [reply code/modules/chemistry/tools/patches.dm:480]: Should also probably check if temp is lower than 270 though.

**Pull request merged**